File: dart-client/lib/model/field_installation_object.dart
part of swagger.api;

class FieldInstallationObject {
  
  String name = null;
  

  List<FieldObjectFields> fields = [];
  

  bool custom = null;
  
  FieldInstallationObject();

  @override
  String toString() {
    return 'FieldInstallationObject[name=$name, fields=$fields, custom=$custom, ]';
  }

  FieldInstallationObject.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    name =
        json['name']
    ;
    fields =
      FieldObjectFields.listFromJson(json['fields'])
;
    custom =
        json['custom']
    ;
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'fields': fields,
      'custom': custom
     };
  }

  static List<FieldInstallationObject> listFromJson(List<dynamic> json) {
    return json == null ? new List<FieldInstallationObject>() : json.map((value) => new FieldInstallationObject.fromJson(value)).toList();
  }

  static Map<String, FieldInstallationObject> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, FieldInstallationObject>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new FieldInstallationObject.fromJson(value));
    }
    return map;
  }
}

File: scalaz-client/src/main/scala/io/swagger/client/api/FieldInstallationObject.scala
package io.swagger.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import FieldInstallationObject._

case class FieldInstallationObject (
  name: String,
fields: Option[List[FieldObjectFields]],
custom: Option[Boolean])

object FieldInstallationObject {
  import DateTimeCodecs._

  implicit val FieldInstallationObjectCodecJson: CodecJson[FieldInstallationObject] = CodecJson.derive[FieldInstallationObject]
  implicit val FieldInstallationObjectDecoder: EntityDecoder[FieldInstallationObject] = jsonOf[FieldInstallationObject]
  implicit val FieldInstallationObjectEncoder: EntityEncoder[FieldInstallationObject] = jsonEncoderOf[FieldInstallationObject]
}
